src/connectionManager.ts

```diff
@@ -0,0 +1,264 @@
+const Connection = require('mongodb-connection-model');
+const DataService = require('mongodb-data-service');
+import * as vscode from 'vscode';
+
+import { createLogger } from './logging';
+import { StatusView } from './views';
+
+const log = createLogger('commands');
+
+function getConnectWebviewContent() {
+  return `<!DOCTYPE html>
+  <html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Connect to MongoDB</title>
+    </head>
+    <body>
+      <img src="https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif" width="300" />
+      <h1>MongoDB Connection Details Form</h1>
+      <h2>Also try connecting with URI.</h2>
+    </body>
+  </html>`;
+}
+
+export default class ConnectionManager {
+  // This is a map of connection instance ids to their connection model.
+  private _connectionConfigs: {
+    [key: string]: any
+  } = {};
+
+  private _currentConnection: any;
+  private _currentConnectionInstanceId: string | null = null;
+
+  private _connecting: boolean = false;
+  private _disconnecting: boolean = false;
+
+  private _statusView: StatusView;
+
+  constructor(_statusView: StatusView) {
+    this._statusView = _statusView;
+  }
+
+  public addMongoDBConnection(): Promise<boolean> {
+    log.info('mdb.connect command called.');
+
+    // Display a message box to the user.
+    vscode.window.showInformationMessage('mdb.connect command run.');
+
+    // Create and show a new connect dialogue webview.
+    const panel = vscode.window.createWebviewPanel(
+      'connectDialogueWebview',
+      'Connect to MongoDB', // Title
+      vscode.ViewColumn.One // Editor column to show the webview panel in.
+    );
+
+    panel.webview.html = getConnectWebviewContent();
+
+    return Promise.resolve(true);
+  }
+
+  public async connectWithURI(): Promise<boolean> {
+    log.info('connectWithURI command called');
+
+    let connectionString;
+    try {
+      connectionString = await vscode.window.showInputBox({
+        value: '',
+        placeHolder: 'e.g. mongodb+srv://username:password@cluster0.mongodb.net/admin',
+        prompt: 'Enter your connection string (SRV or standard)',
+        validateInput: text => {
+          let connectionStringError = null;
+          if (text && text.indexOf('mongodb://') === -1 && text.indexOf('mongodb+srv://') === -1) {
```

Comment:
You can use `Connection.isURI(text)` for this check, And I would use `uri` or `connectionString` instead of `text` to make it more readable what is happening here.